neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/impl/DatabaseIdentifier.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.neo4j.remote.impl;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+/**
+ * Provides all information required to connect to a Neo4j database.
+ *
+ * @author Andrea Boriero &lt;dreborier@gmail.com&gt;
+ * @author Gunnar Morling
+ */
+public class DatabaseIdentifier {
+
+	private static final String PROTOCOL = "http";
+	private static final String SLASH = "/";
+
+	private final String host;
+	private final int port;
+	private final String databaseName;
+	private final String userName;
+	private final String password;
+
+	private final URI serverUri;
+	private final URI databaseUri;
+
+	public DatabaseIdentifier(String host, int port, String databaseName, String userName, String password) throws MalformedURLException, URISyntaxException {
+		this.host = host;
+		this.port = port;
+		this.databaseName = databaseName;
+		this.userName = userName;
+		this.password = password;
+
+		this.serverUri = new URL( PROTOCOL, host, port, "" ).toURI();
```

Comment:
No, no particular reason, this is a copy of an existing class.
 
I think it deserve a separate issue